app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,52 @@
+<div class="generalContainer">
+  <div>
+    <%= link_to "Back to recipes", recipes_path, class: "backButton" %>
+  </div>
+
+  <h1 class="pageTitle"> <%= capitalize_text(@recipe.name) %></h1>
+  <div class="recipeShowGeneralInfo">
+    <p class="preparationTime"><strong>Preparation Time: </strong> <%= @recipe.preparation_time_minutes %></p>
+    <%= render 'shared/change_public', recipe: @recipe %>
+    <p class="cookingTime"><strong>Cooking Time: </strong> <%= @recipe.cooking_time_minutes %></p>
+    <p class="recipeDescription"><%= @recipe.description %></p>
+    <%if recipe_author?(@recipe.user)%>
+      <button ><%= link_to "Generate Shopping List", shopping_list_path %></button>
```

Comment:
When accessing a public recipe, the user should be able to generate a shopping list even if they're not the owner of the recipe. So this button should be visible for all users.